yuniql-platforms/oracle/OracleDataService.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System.Collections.Generic;
+using System.Data;
+using Yuniql.Extensibility;
+using Oracle.ManagedDataAccess.Client;
+using System;
+using System.IO;
+using Yuniql.Extensibility.SqlBatchParser;
+using System.Linq;
+
+namespace Yuniql.Oracle
+{
+    ///<inheritdoc/>
+    public class OracleDataService : IDataService, IMixableTransaction
+    {
+        private string _connectionString;
+        private readonly ITraceService _traceService;
+
+        ///<inheritdoc/>
+        public OracleDataService(ITraceService traceService)
+        {
+            this._traceService = traceService;
+        }
+
+        ///<inheritdoc/>
+        public void Initialize(string connectionString)
+        {
+            this._connectionString = connectionString;
+        }
+
+        ///<inheritdoc/>
+        public bool IsTransactionalDdlSupported => false;
+
+        ///<inheritdoc/>
+        public bool IsSchemaSupported { get; } = true;
+
+        ///<inheritdoc/>
+        public bool IsBatchSqlSupported { get; } = true;
+
+        ///<inheritdoc/>
+        public bool IsUpsertSupported => false;
+
+        ///<inheritdoc/>
+        public string TableName { get; set; } = "__yuniql_schema_version";
+
+        ///<inheritdoc/>
+        public string SchemaName { get; set; }
```

Comment:
I guess there must be default schema in oracle like `public`?

Reply:
Usually, USER is the default schema, eg: if connected with SYS then objects are resolved via SYS unless PUBLIC SYNONYM is created.

Reply:
Interesting :) Now i wonder how yuniql would react to this, I something assume there will always be default schema...

Reply:
we can proxy that to Connection String and get get USER from there. If you just give hints about its usage, then we can check some way to mimic/implement that.

Reply:
Lets try to leave it NULL or empty first and see if it breaks anything. I think as long as you dont use the token '${YUNIQL_SCHEMA_NAME}', it will work just fine